spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -42,17 +47,26 @@
 	@ClassRule
 	public static PubSubEmulator emulator = new PubSubEmulator();
 
+	@Rule
+	public TestRule failOnAbsentEmulator = new TestRule() {
```

Comment:
Why do we need this extra rule?
What if I will forget to copy/paste it everywhere?
Why just single `PubSubEmulator` can't serve us all the requirements?

Reply:
The extra rule is there to provide a user-friendly error message. Otherwise you end up with whatever error the binder chooses to serve when faced with a null constructor parameter.
PubSubEmulator emits a warning, but it's hard to find if it's clobbered by a hundred lines of unrelated error messages.